lms/djangoapps/courseware/views.py

```diff
@@ -862,6 +878,56 @@ def get_cosmetic_display_price(course, registration_price):
         return _('Free')
 
 
+class EnrollStaffView(View):
+    """
+    Display the enroll_staff Html page of about.html if user is not a global staff,
+    course id is in valid format and user shouldn't be enrolled before.
+
+    Arguments:
+     - request    : HTTP request
+     - course_id  : course id
+
+    Returns:
+     -RedirectResponse
+    """
+
+    @method_decorator(require_global_staff)
+    @method_decorator(ensure_valid_course_key)
+    def get(self, request, course_id):
+        """
+        Renders Enroll Staff View
+        """
+        user = request.user
+        course_key = from_string_or_404(course_id)
+        with modulestore().bulk_operations(course_key):
+            course = get_course_with_access(user, 'load', course_key)
+            if not registered_for_course(course, user):
+                context = {'course': course,
+                           'csrftoken': csrf(request)["csrf_token"]}
+                return render_to_response('enroll_staff.html', context)
+
+    @method_decorator(require_global_staff)
+    @method_decorator(ensure_valid_course_key)
+    def post(self, request, course_id):
+        """
+        Enroll and returns the response
+        """
+
+        _next = urllib.quote_plus(request.GET.get('next', 'info'), safe='/:?=')
```

Comment:
@adampalay `quote_plus ` is used here because `course_id` have + sign. `safe` is used beacuse this `'/:?='` needed as it is without encoding.